pagination/tests/test_querystrings.py

```diff
@@ -0,0 +1,38 @@
+from django.http import QueryDict
+
+from .utils import RequestTestCase
+from .. import querystrings
+
+
+class TestQuerystringUpdate(RequestTestCase):
+    def setUp(self):
+        """Create a request and put some GET args in it."""
+        self.request = self.create_request()
+        self.request.GET = QueryDict('page=3&q=search_term')
+
+    def test_unchanged_querystring(self):
+        """Test that the method returns an unchanged request.GET if kwargs is empty."""
+        querystring = querystrings.querystring_update(self.request)
+
+        # Re-parse the result into a dict because the query arguments don't appear in
+        # a predictable order.
+        new_querydict = QueryDict(querystring).dict()
+        expected = {
+            'page': '3',
+            'q': 'search_term',
+        }
+        self.assertDictEqual(new_querydict, expected)
+
+    def test_querystring_update(self):
+        """Test that individual GET arguments are updated or preserved correctly."""
+        search_term = 'a_different_search_term'
+        querystring = querystrings.querystring_update(self.request, q=search_term)
+
+        # Re-parse the result into a dict because the query arguments don't appear in
+        # a predictable order.
+        new_querydict = QueryDict(querystring).dict()
+        expected = {
+            'page': '3',
+            'q': search_term,
+        }
+        self.assertDictEqual(new_querydict, expected)
```

Comment:
Going back to what you said about consistency, basically, if we're going to wind up using `assertDictEqual` for some dicts, there's an argument for using it for all dict comparisons.  Then we might use `assertSequenceEqual` for all list/tuple/etc comparisons by the same logic, and so on.  It's a different convention from what we've been using, but it's more forgiving to errors and no less consistent in its own way.  (It's also a bit more explicit I guess?)